.github/workflows/build-docs.yml

```
@@ -0,0 +1,48 @@
+name: Build docs
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+permissions:
+  contents: write
+
+jobs:
+  build-deploy-docs:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Configure Git Credentials
+        run: |
+          git config user.name github-actions[bot]
+          git config user.email 41898282+github-actions[bot]@users.noreply.github.com
+
+      - uses: actions/setup-python@v5
+        with:
+          python-version: "3.13"
+
+      - run: echo "cache_id=$(date --utc '+%V')" >> $GITHUB_ENV 
+
+      - uses: actions/cache@v4
+        with:
+          key: mkdocs-material-${{ env.cache_id }}
+          path: .cache 
+          restore-keys: |
+            mkdocs-material-
+
+      - name: Install dependencies
+        run: |
+          pip install mkdocs-material mkdocstrings[python] mkdocs-autoapi 
+
+      - name: Build docs in PR
+        if: ${{ github.event_name }} == 'pull_request'
+        run: mkdocs build --clean
+
+      - name: Deploy docs if not PR
+        if: ${{ github.event_name }} != 'pull_request'
+        run: mkdocs gh-deploy --force
```

```suggestion
      - name: Deploy docs
        uses: peaceiris/actions-gh-pages@v4
        if: ${{ github.event_name }} == 'push' && ${{ github.ref }} == 'main'
        with:
          github_token: ${{ secrets.GITHUB_TOKEN }}
          publish_dir: ./site
```

I'm more familiar with this action, and I think it removes the need for the git config stuff above.